src/common/http/HttpClient.cpp

```diff
@@ -7,31 +7,87 @@
 #include "http/HttpClient.h"
 #include "process/ProcessUtils.h"
 
+#include <curl/curl.h>
+
+static struct __curl_init_helper {
+    __curl_init_helper() {
+        curl_global_init(CURL_GLOBAL_ALL);
+    }
+    ~__curl_init_helper() {
+        curl_global_cleanup();
+    }
+}curl_init_helper;
+
+struct CxxCurl {
+    CxxCurl():curl(curl_easy_init()) {}
+    ~CxxCurl() {
+        if (curl) {
+            curl_easy_cleanup(curl);
+        }
+    }
+    bool isValid()const {
+        return curl != nullptr;
+    }
+    const std::string &getResponse()const {
+        return response;
+    }
```

Comment:
Please add a rvalue overload so that the caller has the choice to move the response instead of copy.